libs/designsystem/src/lib/components/chart-wip/chart-wip.component.spec.ts

```diff
@@ -0,0 +1,146 @@
+  const createComponent = createComponentFactory({
+    component: ChartWipComponent,
+    declarations: [ChartWipComponent],
+    componentProviders: [MockProvider(ChartJSService)],
+  });
+
+  beforeEach(() => {
+    spectator = createComponent({});
+    component = spectator.component;
+  });
+
+  it('should create', () => {
+    expect(spectator.component).toBeTruthy();
+  });
+
+  it('should have correct default height', () => {
+    expect(spectator.query('.chart-container')).toHaveComputedStyle({
+      height: '300px',
+    });
+  });
+
+  it('should render chart once after view init', () => {
+    const renderChartSpy = spyOn<any>(component, 'renderChart');
+
+    component.ngAfterViewInit();
+
+    expect(renderChartSpy).toHaveBeenCalledTimes(1);
+  });
+
+  it("should be possible to set the height with the '--kirby-chart-height' CSS custom property", () => {
+    const customHeight = '600px';
+    const element = spectator.query('.chart-container') as HTMLElement;
+    expect(element).not.toHaveComputedStyle({ height: customHeight });
+
+    element.style.setProperty('--kirby-chart-height', customHeight);
+
+    expect(element).toHaveComputedStyle({ height: customHeight });
+  });
+
+  describe("when setting height through the 'height' input property", () => {
+    it('should be possible to use a number', () => {
+      const customHeight = 600;
+      const element = spectator.query('.chart-container') as HTMLElement;
+      expect(element).not.toHaveComputedStyle({ height: `${customHeight}px` });
+
+      spectator.setInput('height', customHeight);
+      spectator.detectChanges();
+
+      expect(element).toHaveComputedStyle({ height: `${customHeight}px` });
+    });
+
+    it('should be possible to use a string', () => {
+      const customHeight = '600px';
+      const element = spectator.query('.chart-container') as HTMLElement;
+      expect(element).not.toHaveComputedStyle({ height: customHeight });
+
+      spectator.setInput('height', customHeight);
+      spectator.detectChanges();
+
+      expect(element).toHaveComputedStyle({ height: customHeight });
+    });
+  });
+
+  describe('when changes occur to', () => {
+    const scenarios = {
+      customOptions: {
+        updateFn: 'updateCustomOptions',
+        newValue: { onClick: () => console.log('testing') },
+      },
+      data: { updateFn: 'updateData', newValue: [1, 2, 3] },
+      dataLabels: { updateFn: 'updateDataLabels', newValue: ['one', 'two', 'three'] },
+      type: { updateFn: 'updateType', newValue: 'bar' },
+      annotations: { updateFn: 'updateAnnotations', newValue: [{ type: 'line' }] },
+    };
+
+    Object.entries(scenarios).forEach(([property, { updateFn, newValue }]) => {
+      describe(`${property}`, () => {
+        it(`should only update ${property}`, () => {
+          const updateFnSpy = spyOn<any>(component, updateFn);
+
+          spectator.setInput(property as any, newValue);
+
+          expect(updateFnSpy).toHaveBeenCalledTimes(1);
```

Comment:
This seems like it checks that the related function has been called once, but not that any of the other functions have not been called.

If that is the intention of the test, I think it should be refactored to reflect that. E.g. have two scenarios `old` and `new` where the component values `component.type` etc. can be compared before and after `setInput`.

Otherwise, I think we should remove 'only' in `it('should only update ${property} ...`

Reply:
Or create multiple spies like line 107-112, and expect all other than the current one to have been called 0 times.

Reply:
I've removed `only` from the description &  created a separate tests that checks that none of the other properties are set